### PR TITLE
Move gathered resources to a SummaryInfo struct

### PR DESCRIPTION
**Pull request by skitt**
Signed-off-by: Stephen Kitt <skitt@redhat.com>

### CODE REVIEW

**Review by Jaanki — APPROVED** [review]

File: pkg/subctl/cmd/gather/logs.go [hunk 1]
```diff
@@ -31,8 +31,8 @@ import (
 	"k8s.io/client-go/kubernetes"
 )
 
-var podLogInfos []logInfo
-var podLogInfo logInfo
+var podLogInfos []LogInfo
+var podLogInfo LogInfo
```
Comment (tpantelis) [inline pkg/subctl/cmd/gather/logs.go]: I assume these globals will be removed as well.

**Review by tpantelis — COMMENTED** [review]
Looks good.

**Review by mangelajo — APPROVED** [review]
oops sorry for the conflicts but it looks good.

File: pkg/subctl/cmd/gather/logs.go [hunk 2]
```diff
@@ -97,20 +97,21 @@ func getLogFromStream(logStream io.ReadCloser) (string, error) {
 	return logs.String(), nil
 }
 
-func writeLogToFile(data, podName string, info Info, fileExtension string) error {
-	fileName := filepath.Join(info.DirName, escapeFileName(info.ClusterName+"_"+podName)+fileExtension)
-	f, err := os.Create(fileName)
+func writeLogToFile(data, podName string, info Info, fileExtension string) (*string, error) {
```
Comment (tpantelis) [inline pkg/subctl/cmd/gather/logs.go]: There's really no reason to return a string pointer - if an error occurs return empty string. 

```suggestion
func writeLogToFile(data, podName string, info Info, fileExtension string) (string, error) {
```

**Review by tpantelis — CHANGES_REQUESTED** [review]
Is the intention to merge this PR or https://github.com/submariner-io/submariner-operator/pull/1260?

**Comment by skitt** [comment]
Merged into #1260.

**Pull request closed without merging**